docs/syntax-reference.rst

```diff
@@ -166,13 +166,15 @@ The supported column data types are:
 -  ``BIGINT``
 -  ``DOUBLE``
 -  ``VARCHAR`` (or ``STRING``)
--  ``ARRAY<ArrayType>`` (JSON and AVRO only. Index starts from 0)
+-  ``ARRAY<ArrayType>`` (JSON and AVRO only. Index starts from 1)
 -  ``MAP<VARCHAR, ValueType>`` (JSON and AVRO only)
 -  ``STRUCT<FieldName FieldType, ...>`` (JSON and AVRO only) The STRUCT type requires you to specify a list of fields.
    For each field you must specify the field name (FieldName) and field type (FieldType). The field type can be any of
    the supported KSQL types, including the complex types ``MAP``, ``ARRAY``, and ``STRUCT``. ``STRUCT`` fields can be
    accessed in expressions using the struct dereference (``->``) operator. See :ref:`operators` for more details.
 
+Note that starting KSQL 5.1 array index starts from 1. For the versions of KSQL before 5.1, array index starts from 0.
```

Comment:
I would put this 'note' inline with the ARRAY type in the list, otherwise users may miss it. (Even if that means changing the layout from a list to a table or something).

Also, need to change the default back to 0 and document the configuration setting that allows users to switch to the new more SQwhelly way of working. (as discussed above)

Reply:
Added an inline note in addition to this paragraph. The default is now 0.